### ISSUE TITLE
allow validation against subschemas

### ISSUE THREAD

**Issue by casualjim** [xeipuuv/gojsonschema#43]
It would be nice to be able to validate a document against a subschema.

something like:

```go
schema.Validate(document, "#/paths/~1pets~1{id}/delete/parameters/0")
```

When executed against a petstore swagger spec it would get the first parameter schema and validate the document against that.

See here for the json:
https://github.com/casualjim/go-swagger/blob/ac3c88ca90cca4d37347603521d29f33246d9827/testing/data.go#L269-L276

**Comment by xeipuuv** [comment]
I see, interesting. I will think about it.

**Comment by haibinzumata** [comment]
When I do validation, id in the resource for creation is not required but id in the resource for updating is required. So I cannot do validation against a resource without linking the resource to a method (POST or PUT). If gojsonschema can do validation against a subschema under LINKS in json schema, it would be very helpful.

**Comment by johnSchnake** [comment]
I have a related PR in that accomplished this for my needs: https://github.com/xeipuuv/gojsonschema/pull/55 Still waiting to hear about desired changes or have it merged though.

**Comment by xeipuuv** [comment]
The issue there is that your pull request feels more like a hack than a feature.
Also there is no documentation (README) and no unit test. 
That explains why it was not merged yet.

**Comment by xeipuuv** [comment]
The issue there is that your pull request feels more like a hack than a feature.
Also there is no documentation (README) and no unit test. 
That explains why it was not merged yet.

**Comment by johnSchnake** [comment]
Agreed; that's why I was trying to discuss the issue, I just never heard back about it.  Thought I'd tack a link into the issue to spur conversation if someone else wanted the feature.

**Comment by xeipuuv** [comment]
Yes you are right, this kind of feature is interesting and deserves some discussion.
I am myself "over super crazy busy" and that explains the lack of time, communication and 
unfortunately manners, so sorry about that ;) 
I hope you guys can figure out the best solution.

Personally, i like what https://github.com/casualjim proposed initially, 
something along the lines of:

```
result, err := gojsonschema.Validate(schemaLoader, documentLoader, referenceToSubSchema)
```
and/or
```
result1, err := schema.Validate(documentLoader1, referenceToSubSchema)
```

So that it respects the interface/usual way of dealing with validations.

Now golang doesnt support overloading and i guess "partial schema" is a correct and better way to qualify that feature (compared to sub schema), so i guess it would become:

```
result, err := gojsonschema.ValidatePartial(schemaLoader, documentLoader, referenceToPartialSchema)
```
and/or
```
result1, err := schema.ValidatePartial(documentLoader1, referenceToPartialSchema)
```

Finally some care has to be taken concerning d.pool.SetStandaloneDocument since it is constrained to some limitations and thats where most of the effort focused to make sure internal and external references to other schemas is respected. This is also where the unit tests must be focused on.

I hope it helps.

**Comment by Igosuki** [comment]
I think it'd be easier if the URL parser could handle segments such as "file:///myschema#/somesegment_to_subschema"
